src/org/esupportail/esupAgent/domain/beans/Avancement.java

```diff
@@ -53,7 +53,7 @@ public Avancement(DonneesAvancementDto da) {
 			dateDernierChangementEstFuture = !d.before(Calendar.getInstance());		
 		}
 		try {
-			echelonDernierChangement = Integer.parseInt(da.getEchelonFuturDto().getCodeEchelonFutur()) - 1;
+			echelonDernierChangement = Integer.parseInt(da.getEchelonFuturDto().getCodeEchelonFutur());
 		} catch (NumberFormatException e) {			
```

Comment:
ce serait plutôt :

|echelonDernierChangement = 
Integer.parseInt(da.getEchelonFuturDto().getCodeEchelonFutur()) - 
(dateDernierChangementEstFuture ? 0 : 1); |



On 17/05/2016 16:49, Pascal Rigaux wrote:
>
> In src/org/esupportail/esupAgent/domain/beans/Avancement.java 
> <https://github.com/EsupPortail/esup-agent/pull/3#discussion_r63536075>:
>
> > @@ -53,7 +53,7 @@ public Avancement(DonneesAvancementDto da) {
> >  			dateDernierChangementEstFuture = !d.before(Calendar.getInstance());		
> >  		}
> >  		try {
> > -			echelonDernierChangement = Integer.parseInt(da.getEchelonFuturDto().getCodeEchelonFutur()) - 1;
> > +			echelonDernierChangement = Integer.parseInt(da.getEchelonFuturDto().getCodeEchelonFutur());
> >  		} catch (NumberFormatException e) {			
>
> Le code me parait bon dans le cas dateDernierChangementEstFuture.
> Par contre effectivement, il ne faudrait peut-être pas faire le -1 
> quand !dateDernierChangementEstFuture.
> Je propose donc :
>
> |echelonDernierChangement = 
> Integer.parseInt(da.getEchelonFuturDto().getCodeEchelonFutur()) - 
> (dateDernierChangementEstFuture ? 1 : 0); |
>
> WYT?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/EsupPortail/esup-agent/pull/3/files/b91888c81fd8bbef749fc36cb1e19d834125454f#r63536075>
>

-- 
========================================
Jean-Pierre TRAN
Assitant Ingénieur
DSI - Pôle système d'information
Université de ROUEN
Bat.16-IRESE-B-Place Emile Blondel
76821 MONT-SAINT-AIGNAN CEDEX
Accès: http://goo.gl/cYgtX

Tél: 02.35.14.71.11
Accueil DSI: 02.35.14.61.00
Fax: 02.35.14.64.64
Mail fonc: si@univ-rouen.fr
Mail pers: jean-pierre.tran@univ-rouen.fr

========================================